discord/automod.py

```diff
@@ -0,0 +1,420 @@
+    rule_trigger_type: :class:`AutoModRuleTriggerType`
+        The trigger type of the rule that was triggered.
+    user_id: :class:`int`
+        The ID of the user that triggered the rule.
+    alert_system_message_id: Optional[:class:`int`]
+        The ID of the system message that was sent to the predefined alert channel.
+    content: :class:`str`
+        The content of the message that triggered the rule.
+    matched_keyword: Optional[:class:`str`]
+        The matched keyword from the triggering message.
+    matched_content: Optional[:class:`str`]
+        The matched content from the triggering message.
+    """
+
+    __slots__ = (
+        '_state',
+        '_guild_id',
+        '_channel_id',
+        'action',
+        'rule_id',
+        'rule_trigger_type',
+        'user_id',
+        'message_id',
+        'alert_system_message_id',
+        'content',
+        'matched_keyword',
+        'matched_content',
+    )
+
+    def __init__(self, *, data: AutoModerationActionExecutionPayload, state: ConnectionState) -> None:
+        self._state: ConnectionState = state
+        self._guild_id: int = int(data['guild_id'])
+        self._channel_id: Optional[int] = utils._get_as_snowflake(data, 'channel_id')
+        self.message_id: Optional[int] = utils._get_as_snowflake(data, 'message_id')
+        self.action: AutoModRuleAction = AutoModRuleAction.from_data(data['action'])
+        self.rule_id: int = int(data['rule_id'])
+        self.rule_trigger_type: AutoModRuleTriggerType = try_enum(AutoModRuleTriggerType, data['rule_trigger_type'])
+        self.user_id: int = int(data['user_id'])
+        self.alert_system_message_id: Optional[int] = utils._get_as_snowflake(data, 'alert_system_message_id')
+        self.content: str = data['content']
+        self.matched_keyword: Optional[str] = data['matched_keyword']
+        self.matched_content: Optional[str] = data['matched_content']
+
+    def __repr__(self) -> str:
+        return f"<AutoModRuleExecution rule_id={self.rule_id} action={self.action!r}>"
+
+    @property
+    def guild(self) -> Guild:
+        """:class:`Guild`: The guild this rule was executed in."""
+        return self._state._get_guild(self._guild_id)  # type: ignore # this should never be None here
+
+    @property
+    def channel(self) -> Optional[Union[GuildChannel, Thread]]:
+        """Optional[Union[:class:`abc.GuildChannel`, :class:`Thread`]]: The channel this rule was executed in."""
+        if self._channel_id:
+            return self.guild and self.guild.get_channel(self._channel_id)
+        return None
+
+    @property
+    def member(self) -> Optional[Member]:
+        """Optional[:class:`Member`]: The member this rule was executed on/who triggered this rule."""
+        return self.guild and self.guild.get_member(self.user_id)
+
+    async def fetch_rule(self) -> AutoModRule:
+        """|coro|
+
+        This method will fetch the rule that was executed.
```

Comment:
```suggestion
        Fetch the rule that was executed.
```

Permissions, raises? The documentation for this needs to be fleshed out.